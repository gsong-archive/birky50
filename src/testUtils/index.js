import * as faker from "faker";

const address = () =>
  [
    faker.address.streetAddress(),
    faker.address.city(),
    `${faker.address.stateAbbr()} ${faker.address.zipCode()}`,
  ].join(", ");

const card = () => ({
  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
  url: faker.internet.url(),
  price: faker.random.word(),
  address: address(),
  addressUrl: faker.internet.url(),
  phone: faker.phone.phoneNumber(),
  imgUrl: faker.image.imageUrl(),
});

const generate = {
  address,
  card,
  phoneNumber: faker.phone.phoneNumber,
  url: faker.internet.url,
  words: faker.random.words,
};

export { generate };
