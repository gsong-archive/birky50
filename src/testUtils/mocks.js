import * as cssFeatures from "../styles/cssFeatures";
import * as styleUtils from "../styles/utils";

export const mockGetBaseFontSize = {
  mock: () => (styleUtils.getBaseFontSize = jest.fn(() => 16)),
  restore: () => styleUtils.getBaseFontSize.mockRestore(),
};

export const mockSupportsGrid = bool => (cssFeatures.supportsGrid = bool);

export const mockSupportsObjectFit = bool =>
  (cssFeatures.supportsObjectFit = bool);
