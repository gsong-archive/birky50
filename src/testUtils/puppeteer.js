import "../env";

const DEFAULT_APP_URL = "http://localhost:3000";
const DEFAULT_JEST_TIMEOUT = 5000;
const DEFAULT_SCREENSHOTS_FOLDER = "screenshots/puppeteer";
const DEFAULT_SCROLL_CHECK_DELAY = 100;

const appUrl =
  process.env.APP_URL === undefined ? DEFAULT_APP_URL : process.env.APP_URL;

const jestTimeOut =
  parseInt(process.env.JEST_TIMEOUT, 10) || DEFAULT_JEST_TIMEOUT;

const screenshotsFolder =
  process.env.PUPPETEER_SCREENSHOTS_FOLDER === undefined
    ? DEFAULT_SCREENSHOTS_FOLDER
    : process.env.PUPPETEER_SCREENSHOTS_FOLDER;

const scrollCheckDelay =
  process.env.PUPPETEER_SCROLL_CHECK_DELAY || DEFAULT_SCROLL_CHECK_DELAY;

export const settings = { appUrl, jestTimeOut, scrollCheckDelay };

export const screenshot = async (imageName, page, options = {}) => {
  await page.screenshot({
    path: `${screenshotsFolder}/${imageName}.png`,
    ...options,
  });
};

const desktops = {
  "macbook-13": { width: 1280, height: 800 },
  "macbook-15": { width: 1440, height: 900 },
};

const mobiles = {
  "ipad-portrait": { width: 768, height: 1024 },
  "iphone-se-portrait": { width: 320, height: 568 },
  "iphone-8-portrait": { width: 375, height: 667 },
  "iphone-8-plus-portrait": { width: 540, height: 960 },
  "iphone-x-portrait": { width: 375, height: 812 },
};

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device.replace("portrait", "landscape")] = {
    width: viewport.height,
    height: viewport.width,
    isLandscape: true,
  };
});

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device] = { ...viewport, isMobile: true, hasTouch: true };
});

export const devices = Object.assign({}, desktops, mobiles);
