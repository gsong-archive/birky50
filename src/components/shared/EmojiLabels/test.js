//@flow strict
import React from "react";

import { render } from "react-testing-library";

import { generate } from "testUtils";

import {
  Airplane,
  Airport,
  Burger,
  Calendar,
  Hotel,
  Location,
  PartyHat,
  Phone,
  WomanBiking,
  WomanRaisingHand,
  EmojiLabel,
} from "./index";

test("Emojis are rendered properly", () => {
  const emojis = [
    [Airplane, "✈️"],
    [Airport, "🛬"],
    [Burger, "🍔"],
    [Calendar, "📆"],
    [Hotel, "🏩"],
    [Location, "📍"],
    [PartyHat, "🎉"],
    [Phone, "📞"],
    [WomanBiking, "🚴🏽‍♀️"],
    [WomanRaisingHand, "🙋🏻‍♀️"],
  ];

  emojis.forEach(([Component, emoji]) => {
    const props = { label: generate.words() };
    const { container, getByText } = render(<Component {...props} />);
    const emojiNode = getByText(emoji);

    expect(container).toHaveTextContent(props.label);
    expect(emojiNode).toHaveTextContent(emoji);
    expect(emojiNode).toHaveAttribute("aria-label");
  });
});

test("EmojiLabel is rendered", () => {
  const props = { emoji: "🦄", ariaLabel: "unicorn", label: "Very Important" };
  const { container } = render(<EmojiLabel {...props} />);

  expect(container).toMatchSnapshot();
});
