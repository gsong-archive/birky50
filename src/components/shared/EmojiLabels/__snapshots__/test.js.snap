// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiLabel is rendered 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-0 {
  font-family: "Segoe UI Emoji";
  margin-right: 0.25em;
}

<div>
  <span
    class="emotion-1"
  >
    <span
      class="emotion-0"
      role="img"
      aria-label="unicorn"
    >
      ðŸ¦„
    </span>
    Very Important
  </span>
</div>
`;
